### ISSUE TITLE
exclude default setting should exclude e2e tests folder

### ISSUE THREAD

**Issue by EgonBolton** [vitest-dev/vitest#3102]
### Clear and concise description of the problem

Since a generally accepted good practice* is to put unit tests next to the code they test, and e2e tests in a separate folder, excluding such a folder would be a better default setting.

In many projects I see that all the configuration file does is modify `include` or `exclude` to exclude e2e tests. This change would allow an additional configuration file to be avoided.

I can do the PR.

### Suggested solution

Add any of the following paths to the default exclude configuration:

**Option 1**: `'**/tests/**'`
This is the most opinionated option. It looks pretty clean, but I can see how some people may have put unit tests inside folders with that name. With this option, they would have to modify the config or rename the folder to something like `/unit-test/`.

**Option 2**: `'**/e2e-tests/**'`
Non-opinionated, and not potentially conflicting. 

**Option 3**:  `'**/tests/**', '**/e2e-tests/**'`
Despite the potential conflict explained in option 1... the goal of default settings is to respond to the most popular settings, so possibly this is also a good option.

### Additional context

*Validated in all the discussions I found on the subject. For example [here](https://www.reddit.com/r/node/comments/kesbv9/comment/gg4tpn7/?utm_source=reddit&utm_medium=web2x&context=3), or [here](https://kit.svelte.dev/docs/project-structure).

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by sheremet-va** [comment]
We should have a discussion about this with the community - we already changed default globs which cause a lot of errors - please, feel free to discuss it in #3530

**Comment by nickmccurdy** [comment]
I feel like this is a different enough situation that it deserves its own discussion/decision. Specifically, e2e tests tend to use proprietary APIs that won't exist and will therefore fail on Vitest. However, Vitest is mostly compatible with Jest's behavior (though the global name is different).

**Comment by GermanJablo** [comment]
Thanks for your response!

@nickmccurdy totally agree! I also reasoned that if the `cypress` folder was being ignored, then `playwright` should be ignored as well.

@sheremet-va The discussion with the community that you speak of, do you say we do it in #3530?
It seems to me that the decision to ignore `test.ts` is more difficult than the decision to ignore `playwright` and `e2e-tests`. Are there any potential risks that you are seeing with this change?

**Comment by sheremet-va** [comment]
> @sheremet-va The discussion with the community that you speak of, do you say we do it in #3530?
> 
> It seems to me that the decision to ignore `test.ts` is more difficult than the decision to ignore `playwright` and `e2e-tests`. Are there any potential risks that you are seeing with this change?
> 
> 

Yes. I'm not a fan of default excluding. It's much harder to remove a folder than to add it. With this change we can cause some tests to silently not run for example.

I don't see why you can't use `e2e-tests` folder to run e2e tests with Vitest's e2e workspace for example.

**Comment by nickmccurdy** [comment]
In that case, wouldn't it be best to remove the `cypress` exclude so Vite is consistent about this? Of course, that would be a breaking change.

**Comment by sheremet-va** [comment]
> In that case, wouldn't it be best to remove the `cypress` exclude so Vite is consistent about this? Of course, that would be a breaking change.

Why? It's not a generic name

**Comment by nickmccurdy** [comment]
> > In that case, wouldn't it be best to remove the `cypress` exclude so Vite is consistent about this? Of course, that would be a breaking change.
> 
> Why? It's not a generic name

Neither is `playwright`, making our e2e test excludes inconsistent, so I'm still in favor of this.

**Comment by silverwind** [comment]
-1 from me for excluding nonstandard folders. For example some repos I contribute to use `tests/e2e`, you can never make it right for everyone.

**Comment by GermanJablo** [comment]
@silverwind Do they have Vitest tests inside `tests/e2e`? If so, could I know why?

**Comment by silverwind** [comment]
In this case `tests/e2e` is playwright, as one might expect.

**Comment by GermanJablo** [comment]
Well, that's the point of this thread. make _`Vitest`_ ignore the e2e folder

**Comment by silverwind** [comment]
It was not me who came up with it, but we do use the `foo.test.e2e.js` pattern for e2e test files, and if this is a common pattern, it might also serve well because vitest's default pattern `**/*.{test,spec}.?(c|m)[jt]s?(x)` does already not match on these files, so no folder-based exclusions would be needed.

**Comment by sheremet-va** [comment]
Vitest should not limit e2e tests as a concept. Vitest is a test runner, you can run unit/integration/e2e tests with Vitest, and no one is stopping you.

I think the whole concept of excluding e2e tests is bad by design. Here we should either add `playwright` to `excludes` and add all new test runners all the time or remove `cypress` and keep it as is. We could also probably remove `dist` folder as we did in other places.

@antfu what do you think?

**Comment by silverwind** [comment]
I tend to agree, a unopinionated default config is best long-term. BTW, what is `.idea` and why do you want to keep it?

**Comment by sheremet-va** [comment]
> I tend to agree, a unopinionated default config is best long-term and it stops these kind of pointless discussions. BTW, what is `.idea` and why do you want to keep it?

It's a folder created by JetBrains IDEs. Since it's standardized (unlike the other three), I decided to keep it. We can also include `.vscode`, but it's usually quite small

**Comment by silverwind** [comment]
Yes, if you exclude one editor folder, you also have to include the others, again making this config opinionated. I think the ideal default exclusions are just `node_modules` and `.git`.

**Comment by antfu** [comment]
I can't remember why we have the config file in exclude but I agree to remove them from default.

For `.cache`, `.temp`, etc, maybe we could ignore all the `.` folders by default, as the naming in Linux already indicates they are "hidden" files.

For cypress am don't have a strong opinion, removing it indeed prevents us from adding more framework specific exclude in the future.

**Comment by GermanJablo** [comment]
Well @sheremet-va, I already got the point about e2e in Vitest, thanks for clearing it up.

Regarding Playwright and Cypress, since I see that some of you do not have strong opinions about it, I want to say that I would prefer that they be excluded by default.

Suppose another test library with a similar popularity emerges in the future. Excluding `playwright` or `cypress` but not that library is not that bad at all. And adding it would take 1 minute (taking this discussion as a starting point).

On balance, I think it's better for DX not to add an extra configuration file.

**Comment by lloydjatkinson** [comment]
The solution I went with and my `package.json`. This way, Vitest isn't picking up the playwright tests (that fail to run in Vitest *anyway* currently).

```js
    test: {
        exclude: [
            '**/node_modules/**',
            '**/integration/playwright/**'
        ],
    },
```

```json
"test:unit": "vitest run",
"test:integration": "playwright test",
"test:integration:ui": "playwright test --ui",
```
